common/src/api/external/mod.rs

```diff
@@ -1258,6 +1304,116 @@ fn label_schema(
     .into()
 }
 
+/// An IP Range is a contiguous range of IP addresses, usually within an IP
+/// Pool.
+#[derive(Clone, Copy, Debug, Deserialize, Serialize)]
+pub enum IpRange {
+    V4(Ipv4Range),
+    V6(Ipv6Range),
+}
+
+impl JsonSchema for IpRange {
```

Comment:
I talked with Adam a bit about this. I think the goal would longer-term be to put the contents of `label_schema` into `schemars` itself, so that we can just `#[derive(JsonSchema)]` again. I'm not sure what to do in the meantime.